cmip6_downscaling/methods/bcsd.py

```diff
@@ -27,10 +24,10 @@ def make_flow_paths(
     PREDICT_PERIOD_START: str,
     PREDICT_PERIOD_END: str,
     VARIABLE: str,
-    LATMIN: str,
-    LATMAX: str,
-    LONMIN: str,
-    LONMAX: str,
+    LATMIN: str,  # Q: why is this unused
+    LATMAX: str,  # Q: why is this unused
+    LONMIN: str,  # Q: why is this unused
+    LONMAX: str,  # Q: why is this unused
```

Comment:
@orianac / @norlandrhagen - I have this question here and elsewhere. I'm probably missing something here so just explain to me what you were trying to accomplished here.

Reply:
I think if parameters are in the xpersist target naming, then they also have to be in the task. 

If I remove `predict_period_start` and `predict_period_end`:

```python

└── 20:31:22 | INFO    | Task 'return_obs': Starting task run...
└── 20:31:22 | ERROR   | Unexpected error: KeyError('predict_period_start')
Traceback (most recent call last):
  File "/srv/conda/envs/notebook/lib/python3.9/site-packages/prefect/engine/runner.py", line 48, in inner
    new_state = method(self, state, *args, **kwargs)
  File "/srv/conda/envs/notebook/lib/python3.9/site-packages/prefect/engine/task_runner.py", line 742, in check_target
    if result.exists(target, **formatting_kwargs):  # type: ignore
  File "/srv/conda/envs/notebook/lib/python3.9/site-packages/xpersist/prefect/result.py", line 153, in exists
    return location.format(**kwargs) in self.cache_store
KeyError: 'predict_period_start'
```

Reply:
i think the goal of specifying the coordinates was to allow for running domain spatial subsets. and it is being used in the `bcsd_flow.py` script currently.

Reply:
could this not be a great place to use the prefect context? so populate the `target_naming_string` template with the prefect context instead of the parameters. that way it can be passed to xpersist as an already-formatted string?

Reply:
I think I'm still failing to get something. The `make_flow_paths` never uses these variables. Or does it somehow and I'm just not seeing where?

Reply:
ooooh!! good catch! yeah, looks like the subset bounding box never made it into that path spec! so to get this to work properly it would need to get added in lines 70-73. but this remains a situation where we might want to switch away from passing these variables around all together....